5_data_munge/src/munge_data_files.R

```diff
@@ -0,0 +1,158 @@
+f_to_c <- function(f) {
+  c <- (f - 32) * (5/9)
+  return(c)
+}
+
+munge_wqp_withdepths <- function(in_ind, min_value, max_value, max_daily_range, out_ind) {
+  dat <- readRDS(sc_retrieve(in_ind)) %>%
+    filter(!is.na(`ResultMeasure/MeasureUnitCode`)) %>%
+    filter(ActivityMediaName %in% 'Water') %>%
+    filter(!is.na(`ActivityDepthHeightMeasure/MeasureValue`)|
+                         !is.na(`ResultDepthHeightMeasure/MeasureValue`)|
+                         !is.na(`ActivityTopDepthHeightMeasure/MeasureValue`)|
+                         !is.na(`ActivityBottomDepthHeightMeasure/MeasureValue`)) %>% 
+    filter(ActivityStartDate > as.Date('1800-01-01') & ActivityStartDate < as.Date('2020-01-01'))
+  
+  # create depth column
+  dat <- mutate(dat, sample_depth = case_when(
+    !is.na(`ActivityDepthHeightMeasure/MeasureValue`) ~ `ActivityDepthHeightMeasure/MeasureValue`,
+    is.na(`ActivityDepthHeightMeasure/MeasureValue`) & !is.na(`ResultDepthHeightMeasure/MeasureValue`) ~ `ResultDepthHeightMeasure/MeasureValue`,
+    is.na(`ActivityDepthHeightMeasure/MeasureValue`) & is.na(`ResultDepthHeightMeasure/MeasureValue`) & !is.na(`ActivityTopDepthHeightMeasure/MeasureValue`) ~ `ActivityTopDepthHeightMeasure/MeasureValue`,
+    is.na(`ActivityDepthHeightMeasure/MeasureValue`) & is.na(`ResultDepthHeightMeasure/MeasureValue`) & is.na(`ActivityTopDepthHeightMeasure/MeasureValue`) ~ `ActivityBottomDepthHeightMeasure/MeasureValue`
+  ))
+  
+  dat <- mutate(dat, sample_depth_unit_code = case_when(
+    !is.na(`ActivityDepthHeightMeasure/MeasureValue`) ~ `ActivityDepthHeightMeasure/MeasureUnitCode`,
+    is.na(`ActivityDepthHeightMeasure/MeasureValue`) & !is.na(`ResultDepthHeightMeasure/MeasureValue`) ~ `ResultDepthHeightMeasure/MeasureUnitCode`,
+    is.na(`ActivityDepthHeightMeasure/MeasureValue`) & is.na(`ResultDepthHeightMeasure/MeasureValue`) & !is.na(`ActivityTopDepthHeightMeasure/MeasureValue`) ~ `ActivityTopDepthHeightMeasure/MeasureUnitCode`,
+    is.na(`ActivityDepthHeightMeasure/MeasureValue`) & is.na(`ResultDepthHeightMeasure/MeasureValue`) & is.na(`ActivityTopDepthHeightMeasure/MeasureValue`) ~ `ActivityBottomDepthHeightMeasure/MeasureUnitCode`
+  ))
+  
+
+  # now reduce to daily measures
+  dat_reduced <- dat %>%
+    filter(ResultMeasureValue > min_value) %>%
```

Comment:
it would probably make sense to move this down below the conversion to deg C, since 31 or 32 deg F would pass this filter but would yield <=0 values after conversion

Reply:
Fixed in two places.